src/app/portlets/content-types/fields/content-type-fields-variables/content-type-fields-variables.component.ts

```diff
@@ -0,0 +1,138 @@
+            ])
+            .subscribe((messages: {[key: string]: string}) => {
+                this.messages = messages;
+                this.initTableData();
+            });
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes.field.currentValue && !changes.field.firstChange) {
+            this.initTableData();
+        }
+    }
+
+    /**
+     * Handle Delete event
+     * @param {FieldVariable} variable
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    deleteVariable(variable: FieldVariable): void {
+        this.fieldVariablesService.deleteFieldVariables(this.field.typeId, this.field.id, variable.id).subscribe(() => {
+            this.fieldVariables = this.fieldVariables.filter((item: FieldVariable) => item !== variable);
+        });
+    }
+
+    /**
+     * Handle Save event
+     * @param {FieldVariable} variable
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    saveVariable(variable: FieldVariable): void {
+        this.fieldVariablesService.saveFieldVariables(this.field.typeId, this.field.id, variable)
+            .subscribe((savedVariable: FieldVariable) => {
+                variable = savedVariable;
+                this.initTableData();
+            });
+    }
+
+    /**
+     * Handle table input's placeholder when an input has been changed succesfully
+     * @param {FieldVariable} variable
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    changeVariable(variable: any): void  {
+        this.handlePlaceholderInput(
+            variable,
+            '',
+            '',
+            this.messages['contenttypes.field.variables.key_placeholder.label'],
+            this.messages['contenttypes.field.variables.value_placeholder.label']);
+    }
+
+    /**
+     * Handle table input's placeholder when an input get focus
+     * @param {FieldVariable} variable
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    changeInitVariable(variable: any): void  {
+        this.handlePlaceholderInput(
+            variable,
+            this.messages['contenttypes.field.variables.key_placeholder.label'],
+            this.messages['contenttypes.field.variables.value_placeholder.label'],
+            '',
+            '');
+    }
+
+    /**
+     * Handle table input's placeholder when an input lost focus
+     * @param {FieldVariable} variable
+     * @memberof ContentTypeFieldsVariablesComponent
+     */
+    changeCancelVariable(variable: any): void  {
+        this.handlePlaceholderInput(
+            variable,
+            '',
+            '',
+            this.messages['contenttypes.field.variables.key_placeholder.label'],
+            this.messages['contenttypes.field.variables.value_placeholder.label']);
+    }
+
+    // tslint:disable-next-line:cyclomatic-complexity
+    private handlePlaceholderInput(variable: any, compareKey: string, compareValue: string, keyLbl: string, valLbl: string): void {
```

Comment:
https://blog.risingstack.com/javascript-clean-coding-best-practices-node-js-at-scale/

<img width="789" alt="screen shot 2018-09-11 at 12 43 20 pm" src="https://user-images.githubusercontent.com/751424/45380588-4b829f00-b5c0-11e8-9090-afc8cdc74170.png">

Reply:
I don't understand waht is this method doing but we should find a simple way to do it (this method is very complex)